model/src/main/java/com/epam/esm/model/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.esm.model.service.impl;
+
+import com.epam.esm.model.dao.UserDao;
+import com.epam.esm.model.dao.entity.User;
+import com.epam.esm.model.service.Page;
+import com.epam.esm.model.service.UserService;
+import com.epam.esm.model.service.converter.impl.UserDTOMapper;
+import com.epam.esm.model.service.dto.UserDTO;
+import com.epam.esm.model.service.exception.IncorrectArgumentException;
+import com.epam.esm.model.service.exception.NotExistEntityException;
+import com.epam.esm.model.service.exception.ServiceException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserDao userDao;
+    private final UserDTOMapper userMapper;
+
+    public UserServiceImpl(UserDao userDao, UserDTOMapper userMapper) {
+        this.userDao = userDao;
+        this.userMapper = userMapper;
+    }
+
+    @Override
+    @Transactional
+    public UserDTO find(Long id) throws ServiceException, NotExistEntityException {
+        try {
+            Optional<User> user = userDao.findById(id);
+            return user.map(userMapper::toDTO)
+                    .orElseThrow(() -> new NotExistEntityException("User with id=" + id + " not exist!"));
+        } catch (DataAccessException e) {
+            log.error("Find user service exception", e);
+            throw new ServiceException("Find user service exception", e);
+        }
+    }
+
+    @Override
+    public List<UserDTO> findAll(int page, int size) throws ServiceException, IncorrectArgumentException {
```

Comment:
what about the @Transactional
also why you do not use the @Transactional for count methods?

Reply:
fix